crates/subspace-networking/src/create.rs

```diff
@@ -458,14 +408,25 @@ async fn build_transport(
 }
 
 // Upgrades a provided transport with a multiplexer and a Noise authenticator.
-fn upgrade_transport<StreamSink>(
-    transport: Boxed<StreamSink>,
+fn upgrade_transport<Output, Error, Listener, ListenerUpgrade, Dial>(
+    transport: impl Transport<
+            Output = Output,
+            Error = Error,
+            Listener = Listener,
+            ListenerUpgrade = ListenerUpgrade,
+            Dial = Dial,
+        > + Send
+        + 'static,
```

Comment:
Let's prefer generics over `impl`, it is easier to read function signature that way